pkgs/top-level/ocaml-packages.nix

```
@@ -2245,7 +2245,21 @@ let
 
         google-drive-ocamlfuse = callPackage ../applications/networking/google-drive-ocamlfuse { };
 
-        hol_light = callPackage ../applications/science/logic/hol_light { };
+        hol_light = callPackage ../applications/science/logic/hol_light (
+          let
+            camlp5_ =
+              if lib.versionAtLeast camlp5.version "8.04.00" then
+                camlp5.overrideAttrs (_: {
+                  version = "8.03.2";
+                  __intentionallyOverridingVersion = true;
+                })
+              else
+                camlp5;
+          in
+          {
+            camlp5 = camlp5_;
+          }
+        );
```

cosmetic, feel free to dismiss


```suggestion
        hol_light = callPackage ../applications/science/logic/hol_light {
          camlp5 =
            if lib.versionAtLeast camlp5.version "8.04.00" then
              camlp5.overrideAttrs {
                version = "8.03.2";
                __intentionallyOverridingVersion = true;
              }
            else
              camlp5;
        };
```